crypto/Makefile

```diff
@@ -3,9 +3,18 @@ CC       ?= clang
 LD       ?= $(CC)
 
 SANFLAGS += -fsanitize=fuzzer
-
 CFLAGS   += $(SANFLAGS)
 LDFLAGS  += $(SANFLAGS)
+
+# TODO gcc as well as older clang versions <= 12 do not support this feature
+$(info "info: using -fsanitize-ignorelist")
+CFLAGS   += -fsanitize-ignorelist=fuzzer/sanitizer_ignorelist.txt
```

Comment:
This is going to break CI asan builds which are currently using gcc. Can you please make this conditional for clang?

Reply:
are currently -> will be when #2052 is merged

Reply:
Note that this setting is only used for custom fuzzer builds that satisfy `ifeq ($(FUZZER),1)`. AFAIK those are currently not touched by the CI.

I've thought about the conditional behavior since "silently" enabling or disabling the ignorelist functionality can have unexpected results for the user, for example by hiding bugs or creating performance differences.

Correctly figuring out if the compiler is clang is an involved process as the `$(CC)` program names vary a lot (for example via honggfuzz compiler aliases) and Makefiles don't have native language support for the necessary version string handling and version number comparison. I've added another commit which should cover this in a _reasonable_ fashion.

Reply:
Ah, sorry, I missed the condition. Since it doesn't affect CI the clang detection logic doesn't have to be included.

Reply:
I think the detection logic is an improvement and makes sense to merge.

Benefits:
* developers who have a clang 14.x behind the `clang` binary otherwise run into the `-Wbitwise-instead-of-logical` issues
* comparing different clang < 13, clang >= 13 versions without Makefile modifications gets easier
* better support for future conditional Makefile behavior related to newly introduced clang functionality